extensions-contrib/kubernetes-overlord-extensions/src/main/java/org/apache/druid/k8s/overlord/execution/PodTemplateSelectStrategy.java

```diff
@@ -44,5 +45,5 @@ public interface PodTemplateSelectStrategy
    * @param task The task for which the Pod template is determined.
    * @return The pod template that should be used to run the task.
    */
-  @NotNull PodTemplate getPodTemplateForTask(Task task, Map<String, PodTemplate> templates);
+  @NotNull Pair<String, PodTemplate> getPodTemplateForTask(Task task, Map<String, PodTemplate> templates);
```

Comment:
For code hygiene and readability, I would advise creating a dedicated class `PodTemplateWithKey` rather than use a `Pair`, as pairs are often difficult to interpret.

Also, the javadoc needs to be updated.

Reply:
updated